addons/sale_management/models/sale_order.py

```diff
@@ -202,8 +202,6 @@ def _onchange_product_id(self):
         if pricelist and product:
             partner_id = self.order_id.partner_id.id
             self.price_unit = pricelist.with_context(uom=self.uom_id.id).get_product_price(product, self.quantity, partner_id)
-        domain = {'uom_id': [('category_id', '=', self.product_id.uom_id.category_id.id)]}
-        return {'domain': domain}
```

Comment:
The static domain is already defined:
https://github.com/odoo/odoo/blob/master/addons/sale_management/models/sale_order.py#L176-L177

Reply:
Ah right, that's probably why, it's just that the dynamic domain was left over.